File: mRemoteV1/CS/Connection/Connection.Protocol.List.cs
// VBConversions Note: VB project level imports
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
// End of VB project level imports


namespace mRemoteNG.Connection.Protocol
{
	public class List : CollectionBase
			{
				
#region Public Properties
public Connection.Protocol.Base this[object Index]
				{
					get
					{
						if (Index is Connection.Protocol.Base)
						{
							return Index;
						}
						else
						{
							return ((Connection.Protocol.Base.Default) (List[System.Convert.ToInt32(Index)]));
						}
					}
				}
				
public new int Count
				{
					get
					{
						return List.Count;
					}
				}
#endregion
				
#region Public Methods
				public Connection.Protocol.Base Add(Connection.Protocol.Base cProt)
				{
					this.List.Add(cProt);
					return cProt;
				}
				
				public void AddRange(Connection.Protocol.Base[] cProt)
				{
					foreach (Connection.Protocol.Base cP in cProt)
					{
						List.Add(cP);
					}
				}
				
				public void Remove(Connection.Protocol.Base cProt)
				{
					try
					{
						this.List.Remove(cProt);
					}
					catch (Exception)
					{
					}
				}
				
				public new void Clear()
				{
					this.List.Clear();
				}
#endregion
			}
}

File: mRemoteV1/Connection/Connection.Protocol.List.vb
Namespace Connection

    Namespace Protocol
        Public Class ProtoList
            Inherits CollectionBase

#Region "Public Properties"

            Default Public ReadOnly Property Items(Index As Object) As Base
                Get
                    If TypeOf Index Is Base Then
                        Return Index
                    Else
                        Return CType(List.Item(Index), Base)
                    End If
                End Get
            End Property

            Public Shadows ReadOnly Property Count As Integer
                Get
                    Return List.Count
                End Get
            End Property

#End Region

#Region "Public Methods"

            Public Function Add(cProt As Base) As Base
                Me.List.Add(cProt)
                Return cProt
            End Function

            Public Sub AddRange(cProt() As Base)
                For Each cP As Base In cProt
                    List.Add(cP)
                Next
            End Sub

            Public Sub Remove(cProt As Base)
                Try
                    Me.List.Remove(cProt)
                Catch ex As Exception
                End Try
            End Sub

            Public Shadows Sub Clear()
                Me.List.Clear()
            End Sub

#End Region
        End Class
    End Namespace

End Namespace